src/main/java/minijava/lexer/SimpleLexer.java

```diff
@@ -0,0 +1,410 @@
+package minijava.lexer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import minijava.MJError;
+
+/** SLL(1) parser style lexer implementation. */
+public class SimpleLexer implements Lexer {
+
+  private LexerInput input;
+  private StringTable stringTable = new StringTable();
+  private Token current = null;
+  private List<Character> characters = new ArrayList<>();
+  private Location location;
+  private List<Token> lookAheadBuffer = new ArrayList<>();
+  private Map<Integer, Terminal> keywordTerms = new HashMap<>();
+  private int numberOfEOFs = 0;
+
+  public SimpleLexer(LexerInput input) {
+    this.input = input;
+    keywordTerms.put(StringTable.BOOLEAN_KEYWORD_ID, Terminal.BOOLEAN);
+    keywordTerms.put(StringTable.INT_KEYWORD_ID, Terminal.INT);
+    keywordTerms.put(StringTable.CLASS_KEYWORD_ID, Terminal.CLASS);
+    keywordTerms.put(StringTable.NEW_KEYWORD_ID, Terminal.NEW);
+    keywordTerms.put(StringTable.RETURN_KEYWORD_ID, Terminal.RETURN);
+    keywordTerms.put(StringTable.THIS_KEYWORD_ID, Terminal.THIS);
+    keywordTerms.put(StringTable.WHILE_KEYWORD_ID, Terminal.WHILE);
+    keywordTerms.put(StringTable.IF_KEYWORD_ID, Terminal.IF);
+    keywordTerms.put(StringTable.ELSE_KEYWORD_ID, Terminal.ELSE);
+    keywordTerms.put(StringTable.TRUE_KEYWORD_ID, Terminal.TRUE);
+    keywordTerms.put(StringTable.FALSE_KEYWORD_ID, Terminal.FALSE);
+    keywordTerms.put(StringTable.PUBLIC_KEYWORD_ID, Terminal.PUBLIC);
+    keywordTerms.put(StringTable.STATIC_KEYWORD_ID, Terminal.STATIC);
+    keywordTerms.put(StringTable.VOID_KEYWORD_ID, Terminal.VOID);
+    keywordTerms.put(StringTable.NULL_KEYWORD_ID, Terminal.NULL);
+    keywordTerms.put(StringTable.STRING_KEYWORD_ID, Terminal.STRING);
```

Comment:
We *could* move this work into a static initializer, however I don't think it's worth the trouble...